watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc.py

```diff
@@ -0,0 +1,460 @@
+    m.fs.EDstack.channel_height.fix(7.1e-4)
+    m.fs.EDstack.cell_width.fix(0.197)
+    m.fs.EDstack.cell_length.fix(1.68)
+
+    # Spacer properties
+    m.fs.EDstack.spacer_porosity.fix(0.83)
+    m.fs.EDstack.spacer_specific_area.fix(10400)
+
+    # Electrochemical properties
+    m.fs.EDstack.electrodes_resistance.fix(0)
+    m.fs.EDstack.current_utilization.fix(1)
+    m.fs.EDstack.diffus_mass.fix(1.6e-9)
+
+    mstat.report_statistics(m)
+
+
+def solve(m, solver=None, tee=True, check_termination=True):
+    if solver is None:
+        solver = get_solver()
+    results = solver.solve(m, tee=tee)
+    if check_termination:
+        assert_optimal_termination(results)
+    return results
+
+
+def initialize_system(m, solver=None):
+    # set up solver
+    if solver is None:
+        solver = get_solver()
+    optarg = solver.options
+    # set scaling factors for state vars and call the 'calculate_scaling_factors' function
+    m.fs.properties.set_default_scaling(
+        "flow_mol_phase_comp", 0.1, index=("Liq", "H2O")
+    )
+    m.fs.properties.set_default_scaling(
+        "flow_mol_phase_comp", 1e2, index=("Liq", "Na_+")
+    )
+    m.fs.properties.set_default_scaling(
+        "flow_mol_phase_comp", 1e2, index=("Liq", "Cl_-")
+    )
+    iscale.set_scaling_factor(m.fs.EDstack.cell_width, 5)
+    iscale.set_scaling_factor(m.fs.EDstack.cell_length, 1)
+    iscale.set_scaling_factor(m.fs.EDstack.cell_pair_num, 0.1)
+    iscale.set_scaling_factor(m.fs.EDstack.voltage_applied, 1)
+    iscale.set_scaling_factor(m.fs.pump0.control_volume.work, 1e1)
+    iscale.set_scaling_factor(m.fs.pump1.control_volume.work, 1e1)
+    iscale.calculate_scaling_factors(m)
+
+    # populate intitial properties throughout the system
+    m.fs.feed.initialize(optarg=optarg)
+    propagate_state(m.fs.arc0)
+    m.fs.sepa0.initialize(optarg=optarg)
+    propagate_state(m.fs.arc1b)
+    m.fs.pump1.deltaP[0].fix(2e5)
+    m.fs.pump1.initialize()
+    m.fs.pump1.deltaP[0].unfix()
+    propagate_state(destination=m.fs.pump0.inlet, source=m.fs.pump1.inlet)
+    m.fs.pump0.deltaP[0].fix(2e5)
+    m.fs.pump0.initialize()
+    m.fs.pump0.deltaP[0].unfix()
+    propagate_state(m.fs.arc1f)
+    propagate_state(m.fs.arc3f)
+    m.fs.EDstack.initialize(optarg=optarg)
+    propagate_state(m.fs.arc4)
+    m.fs.prod.initialize(optarg=optarg)
+    propagate_state(m.fs.arc5)
+    m.fs.prod.initialize(optarg=optarg)
+    m.fs.sepa1.split_fraction[0, "to_conc_in1"].fix(
+        max(2 - value(m.fs.recovery_vol_H2O) ** -1, 1e-8)
+    )
+    m.fs.sepa1.initialize(optarg=optarg)
+    m.fs.sepa1.split_fraction[0, "to_conc_in1"].unfix()
+    propagate_state(m.fs.arc6)
+    m.fs.disp.initialize()
+    propagate_state(m.fs.arc7)
+    propagate_state(m.fs.arc3b, direction="backward")
+    propagate_state(m.fs.arc2)
+    m.fs.mix0.initialize(optarg=optarg)
+    m.fs.costing.initialize()
+
+    iscale.calculate_scaling_factors(m)
+
+
+def display_model_metrics(m):
```

Comment:
Curious why you don't use the unit model's report method within this function. It may not cover everything but should cover some key unit-level metrics at least (if you included in the unit model)

Reply:
done.